app/controllers/providers/correspondence_address/care_ofs_controller.rb

```diff
@@ -0,0 +1,36 @@
+module Providers
+  module CorrespondenceAddress
+    class CareOfsController < ProviderBaseController
+      prefix_step_with :correspondence_address
+
+      def show
+        @form = Addresses::CareOfForm.new(model: address)
+      end
+
+      def update
+        @form = Addresses::CareOfForm.new(form_params)
+
+        render :show unless save_continue_or_draft(@form)
+      end
+
+    private
+
+      def legal_aid_application
```

Comment:
I'm not sure this is needed, a legal_aid_application variable in one of the lover level controllers, if you need it to get the applicant you could replace it with 

```ruby
    def applicant
      @applicant ||= @legal_aid_application.applicant
    end
```
or, if you only need the applicant for the address?

```ruby
    def applicant
      @address ||= @legal_aid_application.applicant.address
    end
```

Reply:
👍 updated